src/restricted-endpoints/document/document.controller.ts

```diff
@@ -44,34 +45,42 @@ export class DocumentController {
   ) {}
 
   /**
-   * Fetch a document from a database with basic auth.
-   * See {@link https://docs.couchdb.org/en/stable/api/document/common.html?highlight=put%20document#get--db-docid}
+   * Check document meta information
+   * See {@link https://docs.couchdb.org/en/stable/api/document/common.html#head--db-docid}
    * @param req express Request object
    * @param db the name of the database from which the document should be fetched
    * @param docId the name of the document
    * @param user logged in user
    * @param queryParams additional params that will be forwarded
    */
   @Head()
-  async headDocument(
+  headDocument(
     @Request() req: Req,
     @Param('db') db: string,
     @Param('docId') docId: string,
     @User() user: UserInfo,
     @Query() queryParams?: any,
-  ): Promise<DatabaseDocument> {
+  ): Observable<any> {
     const userAbility = this.permissionService.getAbilityFor(user);
-    const document = await firstValueFrom(
-      this.couchdbService.get(db, docId, queryParams),
-    );
-    if (userAbility.can('read', document)) {
-      if (document._rev) {
-        req.res.setHeader('etag', document._rev);
-      }
-      return document;
-    } else {
+
+    if (
+      !userAbility.can('read', {
+        _id: docId,
+      })
+    ) {
```

Comment:
@tomwwinter If we have a permission rule like "can read all docs with location=Berlin", won't this return 401 even for allowed docs?

Where is the HEAD request used? How critical is a possible difference between the head and actual get?

Reply:
Good point, the field permissions are a litte bit tricky... lets discuss this today